controllers/machinedeletion_controller.go

```diff
@@ -63,19 +73,54 @@ func (r *MachineDeletionReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	if remediation = r.getRemediation(ctx, req); remediation == nil {
 		return ctrl.Result{}, nil
 	}
-	//not a machine based remediation
-	machineOwnerRef := getMachineOwnerRef(remediation)
-	if machineOwnerRef == nil {
-		return ctrl.Result{}, nil
+
+	var machine *unstructured.Unstructured
+	//Health check was done by NHC
+	if node, err := r.getNodeFromMdr(remediation); err == nil {
```

Comment:
my preferred style is to avoid:
```
if err == nil {
   //do lots of work
} else {
  //just return
}
```

I prefer to avoid the redundant `else`, so I don't need to think about it when reading the code:
```
if err != nil {
   //just return
} 
//do lots of work
```
up to you of course